relayer/src/chain/cosmos.rs

```diff
@@ -449,10 +452,15 @@ impl CosmosSdkChain {
     /// The actual gas cost, when a transaction is executed, may be slightly higher than the
     /// one returned by the simulation.
     fn apply_adjustment_to_gas(&self, gas_amount: u64) -> u64 {
-        min(
-            gas_amount + mul_ceil(gas_amount, self.gas_adjustment()),
-            self.max_gas(),
-        )
+        debug_assert!(self.gas_adjustment() <= 1.0);
```

Comment:
Agreed, so as this will break functionality if not upheld, it should be a runtime assertion in release mode as well.

```suggestion
        assert!(self.gas_adjustment() <= 1.0);
```